x-pack/solutions/observability/plugins/observability_agent_builder/server/attachments/log.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { z } from '@kbn/zod';
+import dedent from 'dedent';
+import type { CoreSetup, Logger } from '@kbn/core/server';
+import type { AttachmentTypeDefinition } from '@kbn/onechat-server/attachments';
+import { ToolType } from '@kbn/onechat-common';
+import { ToolResultType } from '@kbn/onechat-common/tools/tool_result';
+import type {
+  ObservabilityAgentBuilderPluginStart,
+  ObservabilityAgentBuilderPluginStartDependencies,
+} from '../types';
+import type { ObservabilityAgentBuilderDataRegistry } from '../data_registry/data_registry';
+import { OBSERVABILITY_LOG_ATTACHMENT_TYPE_ID } from '../../common';
+import { getLogDocumentById } from '../routes/ai_insights/get_log_document_by_id';
+
+const GET_LOG_DETAILS_TOOL_ID = 'get_log_details';
+
+const logDataSchema = z.object({
+  id: z.string(),
+  index: z.string(),
+});
+
+export type LogAttachmentData = z.infer<typeof logDataSchema>;
+
+export function createLogAttachmentType({
+  core,
+  logger,
+  dataRegistry,
+}: {
+  core: CoreSetup<
+    ObservabilityAgentBuilderPluginStartDependencies,
+    ObservabilityAgentBuilderPluginStart
+  >;
+  logger: Logger;
+  dataRegistry: ObservabilityAgentBuilderDataRegistry;
+}): AttachmentTypeDefinition<typeof OBSERVABILITY_LOG_ATTACHMENT_TYPE_ID, LogAttachmentData> {
+  return {
+    id: OBSERVABILITY_LOG_ATTACHMENT_TYPE_ID,
+    validate: (input) => {
+      const parsed = logDataSchema.safeParse(input);
+      if (parsed.success) {
+        return { valid: true, data: parsed.data };
+      }
+      return { valid: false, error: parsed.error.message };
+    },
+    format: (attachment) => {
+      const { index, id } = attachment.data;
+
+      return {
+        getRepresentation: () => ({
+          type: 'text',
+          value: `Observability Log ID: ${index}:${id}. Use the ${GET_LOG_DETAILS_TOOL_ID} tool to fetch full log information.`,
+        }),
+        getBoundedTools: () => [
+          {
+            id: GET_LOG_DETAILS_TOOL_ID,
+            type: ToolType.builtin,
+            description: `Fetch the full log information for log ${index}:${id}.`,
+            schema: z.object({}),
+            handler: async (_args, context) => {
+              try {
+                const [coreStart] = await core.getStartServices();
+                const esClient = coreStart.elasticsearch.client.asScoped(context.request);
+
+                const logEntry = await getLogDocumentById({
+                  esClient: esClient.asCurrentUser,
+                  index,
+                  id,
+                });
+
+                return {
+                  results: [
+                    {
+                      type: ToolResultType.other,
+                      data: logDetails,
```

Comment:
is this meant to be `logEntry `?
```suggestion
                      data: logEntry,
```